### PR TITLE
No bug - More frontend cleanups.

### PR DESCRIPTION
**Pull request by emilio**
See individual commits for details.

### CODE REVIEW

**Comment by emilio** [comment]
This is up in the dev instance, fwiw.

File: static/js/panel.js [hunk 1]
```diff
@@ -1,147 +1,106 @@
...
+    this.rawNode = this.findLink("Raw");
 
-  var permalinkNode = $('.panel a[title="Permalink"]');
-  permalinkNode = permalinkNode.length == 0 ? null : permalinkNode[0];
+    this.toggleButton.addEventListener("click", () => this.toggle());
+    this.accelEnabledCheckbox.addEventListener("change", () => {
+      localStorage.setItem("accel-enable", event.target.checked ? "1" : "0");
+      this.updateAccelerators();
+    });
+    document.documentElement.addEventListener("keypress", event =>
```
Comment (staktrace) [inline static/js/panel.js:18]: Note to self: verified on dev.searchfox.org that this still plays well with typeahead find (i.e. doesn't regress bug 1494348).

File: static/js/context-menu.js [hunk 2]
```diff
@@ -208,31 +208,31 @@ function getTargetWord() {
...
     return null;
   }
 
-  if (endIndex <= startIndex) {
+  if (end <= start) {
     return null;
   }
 
-  return selectedTxtString.substr(startIndex, endIndex - startIndex);
+  return string.substr(start, end);
```
[the remark this answers is not in the document]
Comment (emilio) [reply static/js/context-menu.js:237]: Nice catch! Indeed.

File: static/js/dxr.js [hunk 3]
```diff
@@ -542,6 +542,6 @@ function formatDate(dateString) {
   return date + " " + time;
 }
 
-$(".pretty-date").each(function () {
-  $(this).text(formatDate($(this).data("datetime")));
-});
+for (let element of document.querySelectorAll(".pretty-date")) {
+  element.innerText = formatDate(element.getAttribute("data-datetime"));
```
[the remark this answers is not in the document]
Comment (emilio) [reply static/js/dxr.js:545]: Yeah, I tend to prefer the explicit getAttribute, but no big deal either way.

File: static/js/dxr.js [hunk 4]
```diff
@@ -1,632 +1,565 @@
...
-  /**
-   * Hang an advisory message off the search field.
-   * @param {string} level - The seriousness: 'info', 'warning', or 'error'
-   * @param {string} html - The HTML message to be displayed
-   */
-  function showBubble(level, html, which) {
-    // If hideBubble() was already called, abort the hide animation:
-    $(".bubble").stop();
+  startSearchSoon() {
+    this.cancel();
```
[the remark this answers is not in the document]
Comment (emilio) [reply static/js/dxr.js]: Yeah, the old code would cancel the history query and such, but not ongoing searches. I followed your suggestion here. It wouldn't be _too_ much effort, but I've left as-is for now.

[the remark this answers is not in the document]
Comment (emilio) [reply static/js/dxr.js]: As is as in "one fetch cancels the previous one", but I addopted your suggestion above effectively.

**Pull request merged**